packages/customize-widgets/src/filters/widget-box-parent-support.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+
+import { addFilter } from '@wordpress/hooks';
+
+/**
+ * Removes the "parent" property from the Widget Box block
+ * so as to allow it to appear in the Widgets Customizer.
+ * By default this block is set to appear only as a child
+ * of "widget-area" so as to avoid nesting of Widget Boxes.
+ * However as the Customizer doesn't utilise the `widget-area`
+ * block, we must remove the parent setting.
+ *
+ * @param {Object} settings the block settings.
+ * @param {string} name     the name of the block.
+ * @return {Object} the new block settings.
+ */
+function removeWidgetBoxParent( settings, name ) {
+	if ( name !== 'core/widget-box' ) {
+		return settings;
+	}
+
+	// Remove the "parent" property.
+	// eslint-disable-next-line no-unused-vars
+	const { parent, ...newSettings } = settings;
```

Comment:
An alternative would be

```suggestion
	const newSettings = { ...settings };
	delete newSettings.parent;
```

So that we don't need to disable eslint.

Reply:
However, don't we also want to avoid nesting Widget Boxes here in the customizer too? I don't if there's an official way to do that, but since that the _top-level block_'s name is `null`, seems like `parent: [ null ]` works. Not very readable though.

Reply:
> don't we also want to avoid nesting Widget Boxes here in the customizer too

Yes I'm just not sure how.

> but since that the top-level block's name is null, seems like parent: [ null ] works. Not very readable though.

Oh that `null` thing is strange. Is that reliable or a bit of a hack? 

I guess `InnerBlocks` has an `allowedBlocks` setting which we could use to include all blocks available int he Widgets Editor except `core/widget-box`. 

Perhaps we need a `disallowedBlocks` inverse setting on `InnerBlocks`?

Reply:
I'm wondering if we should be bothered stopping people from nesting Widget Boxes at all. The worst that happens if they do is they get two headings one after the other, which is unsemantic, but is already possible to do with regular blocks if someone really wants to. Trying to prevent this might be more trouble than it's worth 😅

Reply:
If you feel this is ok then that's fine. I've removed the restriction thus simplifying the code.